### ISSUE TITLE
how to use the s3im function on other training processes

### ISSUE THREAD

**Issue by jialong2023** [Madaoer/S3IM-Neural-Fields#6]
very nice job!
I tried to use the s3im loss function on this repo https://github.com/yenchenlin/nerf-pytorch by simply adding the s3im_func:
        _parameter:
        s3im_func = S3IM(kernel_size=4, stride=4, repeat_time=10,
                            patch_height=32, patch_width=32).cuda()_
        ........
        _my code:
        _optimizer.zero_grad()
        img_loss = img2mse(rgb, target_s)
        img_loss += s3im_func(rgb, target_s)
        img_loss.backward()__
I compared the fern data and lego data with or without s3im loss, but the result shows that adding the s3im_func loss got worse performance. Especially, it can't reconstruct the Lego data with the same iteration. Do I misunderstand the s3im_func loss or can't use it in the data with a white background? Can you give any suggestions on how to use s3im_func on other training processes?
![000](https://github.com/Madaoer/S3IM-Neural-Fields/assets/39518637/f12cb337-cbce-4a7d-b1ff-d36f75382107)
without s3im_func on lego
![000](https://github.com/Madaoer/S3IM-Neural-Fields/assets/39518637/70b8677f-62ce-4177-9b07-2e87cb2c823e)
with s3im_func on lego
![000](https://github.com/Madaoer/S3IM-Neural-Fields/assets/39518637/ba2f0f3a-78b0-4d58-b481-60ae0e1ced34)
with s3im_func on fern
![000](https://github.com/Madaoer/S3IM-Neural-Fields/assets/39518637/261d27fc-2fbb-4570-ad68-337fb00e8331)
without s3im_func on fern

**Comment by Madaoer** [comment]
1. I think you can try to finetune the weight of s3im loss, like 0.1. For nerf-pytorch, it seems that the training could fail sometimes [nerf-pytorch issue](https://github.com/yenchenlin/nerf-pytorch/issues/52),  maybe you have to train several times before the training PSNR is reliable.
2. S3IM is more efficient in difficult and complicate scene, you can try the dataset mentioned in NeRFW

**Comment by asher-bit** [comment]
I have the same problem when trying use S3IM in masked_wavelet_transform, which imply in tensorf. It seems like there is nothing happen except the train time up.
I use it in chair scene.Maybe more effective in some complex indoor scenes?

**Comment by Madaoer** [comment]
> I have the same problem when trying use S3IM in masked_wavelet_transform, which imply in tensorf. It seems like there is nothing happen except the train time up. I use it in chair scene.Maybe more effective in some complex indoor scenes?

I think so.
